app/src/main/java/zapsolutions/zap/HomeActivity.java

```diff
@@ -354,43 +357,11 @@ public void onWalletLoadedUpdated(boolean success, int error) {
         } else {
             if (error == Wallet.WalletLoadedListener.ERROR_LOCKED) {
 
-
-                // Show unlock dialog
-                AlertDialog.Builder adb = new AlertDialog.Builder(this);
-                adb.setTitle(R.string.unlock_wallet);
-                adb.setCancelable(false);
-                View viewInflated = LayoutInflater.from(this).inflate(R.layout.dialog_input_password, null, false);
-
-                final EditText input = viewInflated.findViewById(R.id.input);
-                input.setShowSoftInputOnFocus(true);
-                input.requestFocus();
+                if (mUnlockDialog != null && !mUnlockDialog.isShowing()) {
```

Comment:
Now we can check if a unlock dialog is already shown, and don't show a second one.
The bug can be tested if you have a locked lnd instance, and then try to connect, which will open the dialog.
If you then put the app to background and back to foreground, a second dialog will be opened (you can click cancel 2 times).